tests/Unit/PointwiseFunctions/AnalyticSolutions/GrMhd/TestFunctions.py

```diff
@@ -166,3 +166,176 @@ def alfven_dt_magnetic_field(x, t, wavenumber, pressure, rest_mass_density,
 
 
 # End functions for testing AlfvenWave.cpp
+
+
+# Functions for testing BondiMichelMHD.cpp
+
+
+def sonic_fluid_speed_squared(r_c, mass):
+    return 0.5 * mass / r_c
+
+
+def sonic_sound_speed_squared(u_c_2):
+    return u_c_2 / (1.0 - 3.0 * u_c_2)
+
+
+def sonic_newtonian_sound_speed_squared(g, c_s_c_2):
+    return (g - 1.0) * c_s_c_2 / (g - 1.0 - c_s_c_2)
+
+
+def adiabatic_constant(n_s_c_2, g, rho_c):
+    return n_s_c_2 * pow(rho_c, 1.0 - g) / g
+
+
+def mass_accretion_rate_over_four_pi(r_c, rho_c, u_c_2):
+    return r_c**2.0 * rho_c * np.sqrt(u_c_2)
+
+
+def bernoulli_constant_squared_minus_one(n_s_c_2, u_c_2, g):
+    return -3.0 * u_c_2 * (1.0 + n_s_c_2/(g - 1.0))**2 + \
+      (n_s_c_2 / (g - 1.0) * (2.0 + n_s_c_2 / (g - 1.0)))
+
+
+def bernoulli_equation_lhs_squared_minus_one(rho, r, g, k, m_dot_over_four_pi, mass):
+    u = m_dot_over_four_pi / (r**2 * rho)
+    g_minus_one = g - 1.0
+    #polytropic_index times newtonian_sound_speed_squared:
+    ncns2 = k * g * pow(rho, g_minus_one) / g_minus_one
+    #specific enthalpy squared minus one:
+    h2_minus_one = ncns2 * (2.0 + ncns2)
+    h2 = h2_minus_one + 1.0
+    return h2_minus_one + h2 * (-2.0 * mass / r + u**2)
+
+def bernoulli_root_function(rho, r, g, k, m_dot_over_four_pi, n_s_c_2, u_c_2, mass):
+    return bernoulli_equation_lhs_squared_minus_one(rho, r, g, k, m_dot_over_four_pi, mass) \
+      - bernoulli_constant_squared_minus_one(n_s_c_2, u_c_2, g)
+
+def sound_speed_at_infinity_squared(g, c_s_c_2):
+    g_minus_one = g - 1.0
+    return g_minus_one + (c_s_c_2 - g_minus_one) * np.sqrt(1.0 + 3.0 * c_s_c_2)
+
+
+def rest_mass_density_at_infinity(g, rho_c, c_s_inf_2, c_s_c_2):
+    return rho_c * pow(c_s_inf_2/(c_s_c_2 * np.sqrt(1.0 + 3.0 * c_s_c_2)), 1.0/(g-1.0))
+
+def rest_mass_density_at_horizon(g, rho_c, u_c_2):
+    return 0.0625 * rho_c * pow(u_c_2, -1.5)
+
+
+def one_over_lapse(mass, radius):
```

Comment:
I'm not sure this is a good name without a prefix since other solutions could also have a lapse/inverse lapse.